<?xml version="1.0"?>
<launch>

  <node name="transform_republish_node" pkg="sdf_mp_integration" type="transform_republish_node"/>
  
  <node name="hsr_python_controller_node" pkg="sdf_mp_integration" type="hsr_python_controller.py" output="screen"/>

  <param name="transform_topic" type="string" value="transform_topic" />
  <param name="base_goal_sub_topic" type="string" value="/move_base_simple/goal" />
  <param name="pcl_topic" type="string" value="/self_filtered_cloud" />
  <param name="global_frame" type="string" value="odom" />
  <param name="person_positions" type="string" value="/person_positions" />


  <param name="map_size_x" type="int" value="256" />
  <param name="map_size_y" type="int" value="256" />
  <param name="map_size_z" type="int" value="32" />
  <param name="head_behaviour" type="int" value="0" />
  <param name="t_look_ahead" type="int" value="5" />

  <param name="sensor_frame" type="string" value="head_rgbd_sensor_rgb_frame"/>
  <param name="use_rgb" type="bool" value="0" />
  <param name="max_queue_size" type="int" value="1"/>

  <param name="epsilon" type="double" value="0.7" />
  <param name="cost_sigma" type="double" value="0.1" />
  <param name="flag_vel_limit" type="bool" value="true" />

  <!-- Run the robot self filter  -->

  <node pkg="nodelet" type="nodelet" name="mark_nodelet_manager" args="manager" />

  <node pkg="nodelet" type="nodelet" name="nodelet1" args="load depth_image_proc/point_cloud_xyz mark_nodelet_manager">
    <remap from="camera_info" to="/centermask2_depth_ros/camera_info"/>
    <remap from="image_rect" to="/centermask2_depth_ros/static_depth_image"/>
    <remap from="points" to="/local_points"/>
    <param name="queue_size" value="1"/>
  </node>

  <include file="$(find robot_self_filter)/launch/filter.launch">
    <arg name="cloud_in" value="/local_points"/>
    <arg name="cloud_out" value="/self_filtered_cloud"/>
  </include>

  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find sdf_mp_integration)/config/hsr_with_prediction_vis.rviz" />

</launch>